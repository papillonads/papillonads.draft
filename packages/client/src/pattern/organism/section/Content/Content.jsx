import React from 'react'
import cn from 'classnames'
import cs from './Content.scss'
import Banner from '../../../atom/content/Banner'
import LeftColumn from '../../../molecule/content/LeftColumn'
import { contentDefaultProps, contentPropTypes } from './Content.props'

const Content = ({ className, data }) => {
  const contentClassList = cn(className, cs.content)

  return (
    <div className={contentClassList}>
      <div className={cn(cs.content__maxWidth)}>
        <Banner className={cs.content__topBanner} data={data.topBanner} />
        <LeftColumn className={cs.content__leftColumn} data={data.leftColumn} />
      </div>
    </div>
  )
}

Content.defaultProps = contentDefaultProps

Content.propTypes = contentPropTypes

export default Content
