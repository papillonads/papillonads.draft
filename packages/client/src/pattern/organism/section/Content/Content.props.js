import { string, shape } from 'prop-types'
import { bannerPropTypes } from '../../../atom/content/Banner/Banner.props'

export const contentDefaultProps = {
  className: '',
}

export const contentPropTypes = {
  className: string,
  data: shape({
    topBanner: bannerPropTypes.data,
    marketingBanner: bannerPropTypes.data,
  }).isRequired,
}
