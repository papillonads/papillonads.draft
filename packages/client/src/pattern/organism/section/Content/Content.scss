.maxWidth {
  display: flex;
  justify-content: space-between;
  flex-direction: column;

  @media (min-width: 950px) {
    flex-direction: row-reverse;
  }

  padding: 20px 0;
  margin: 0 auto;
  max-width: 1010px;
  min-width: 288px;
}

.topBanner {
  width: 100%;
  padding-left: 24px;

  @media (min-width: 950px) {
    width: 81.7%;
  }
}

.leftColumn {
  min-width: 192.8px;
  width: 19.28%;
  margin-top: 20px;

  @media (min-width: 950px) {
    margin-top: 0;
  }
}
