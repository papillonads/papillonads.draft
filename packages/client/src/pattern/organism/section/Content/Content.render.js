import React from 'react'
import Content from './Content'
import homePageData from '../../../../../../server/src/datasources/page/home'

const defaultRender = <Content data={homePageData.content} />

export { defaultRender }
