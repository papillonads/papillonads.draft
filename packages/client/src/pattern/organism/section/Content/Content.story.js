import { storiesOf } from '@storybook/react'
import { defaultRender } from './Content.render'
import { getInfo } from '../../../../../../../.storybook/library'

const defaultRenderInfo = getInfo('default')

storiesOf('Patttern/Organism/Section/Content', module).add(defaultRenderInfo.text, () => defaultRender, defaultRenderInfo.parameters)
