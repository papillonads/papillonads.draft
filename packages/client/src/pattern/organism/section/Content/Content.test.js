import { defaultRender } from './Content.render'

describe('<Content />', () => {
  let mathFloorSpy

  beforeEach(() => {
    mathFloorSpy = jest.fn(() => {
      return 0
    })
    jest.spyOn(Math, 'floor').mockImplementation(mathFloorSpy)
  })

  afterEach(() => {
    jest.clearAllMocks()
  })

  describe('Snaphot', () => {
    test('must match defaultRender', () => {
      expect(global.renderToJSON(defaultRender)).toMatchSnapshot()
    })
  })
})
