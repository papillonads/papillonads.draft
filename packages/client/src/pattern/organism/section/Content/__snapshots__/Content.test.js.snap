// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Content /> Snaphot must match defaultRender 1`] = `
<div
  className=""
>
  <div
    className="banner"
    style={
      Object {
        "height": "10em",
        "maxWidth": "730px",
      }
    }
  >
    <iframe
      src="https://s0.2mdn.net/4270210/1556116689187/KPN ZAKELIJKE MARKT_Z_IPHONE_728x90/KPN ZAKELIJKE MARKT_Z_IPHONE_728x90.html"
      title="banner"
    />
  </div>
</div>
`;
